config/merge_config.go

```diff
@@ -0,0 +1,377 @@
+package config
+
+import (
+	"fmt"
+	"io"
+	"net/url"
+	"os"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	ddconfig "github.com/DataDog/datadog-agent/pkg/config"
+	ddutil "github.com/DataDog/datadog-agent/pkg/util"
+	"github.com/DataDog/datadog-process-agent/util"
+	log "github.com/cihub/seelog"
+	"github.com/go-ini/ini"
+)
+
+func mergeCheckIntervalIfSet(dc ddconfig.Config, agentConf *AgentConfig, key, mapKey string) {
+	if dc.IsSet(key) {
+		agentConf.CheckIntervals[mapKey] = time.Duration(dc.GetInt(key)) * time.Second
+	}
+}
+
+func mergeString(dc ddconfig.Config, key string, defaultVal string) string {
```

Comment:
can we call these get{String/Int/Bool}OrDefault -- elsewhere in the code `merge` implies that the parameters are being mutated